pkg/manager/manager.go

```diff
@@ -275,7 +281,19 @@ func newControllerManagerWithRetry(cfg *rest.Config, controllerOptions ctrl.Opti
 		localRetry,
 		func(_ error) bool { return true },
 		func() error {
+			// Check if context is cancelled before attempting retry
+			select {
+			case <-ctx.Done():
+				logger.GetLogger().Warn("retry operation cancelled", logfields.Error, ctx.Err())
+				return ctx.Err()
+			default:
+			}
+
 			attempts++
+			// Wrap attempts counter to prevent integer overflow
+			if attempts > math.MaxInt32 {
+				attempts = 0
+			}
```

Comment:
So let's say we do `math.MaxInt32` attempts, the next one will have no delay, and then we will restart the exponential backoff ladder, right?